admin/server-timing.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+ * Server-Timing API admin integration file.
+ *
+ * @package performance-lab
+ */
+
+/**
+ * Adds the Server-Timing page to the Tools menu.
+ *
+ * @since n.e.x.t
+ */
+function perflab_add_server_timing_page() {
+	$hook_suffix = add_management_page(
+		__( 'Server-Timing', 'performance-lab' ),
+		__( 'Server-Timing', 'performance-lab' ),
+		'manage_options',
+		PERFLAB_SERVER_TIMING_SCREEN,
+		'perflab_render_server_timing_page'
+	);
+
+	// Add the following hooks only if the screen was successfully added.
+	if ( false !== $hook_suffix ) {
+		add_action( "load-{$hook_suffix}", 'perflab_load_server_timing_page' );
+	}
+
+	return $hook_suffix;
+}
+add_action( 'admin_menu', 'perflab_add_server_timing_page' );
+
+/**
+ * Initializes settings sections and fields for the Server-Timing page.
+ *
+ * @since n.e.x.t
+ */
+function perflab_load_server_timing_page() {
+	add_settings_section(
+		'benchmarking',
+		__( 'Benchmarking', 'performance-lab' ),
+		static function() {
+			?>
+			<p>
+				<?php esc_html_e( 'In this section, you can provide hook names to include measurements for them in the Server-Timing header.', 'performance-lab' ); ?>
+				<br>
+				<?php
+				echo wp_kses(
+					__( 'For any hook name provided, the <strong>cumulative duration between all callbacks</strong> attached to the hook is measured, in milliseconds.', 'performance-lab' ),
+					array( 'strong' => array() )
+				);
+				?>
+			</p>
+			<?php
+		},
+		PERFLAB_SERVER_TIMING_SCREEN
+	);
+
+	/*
+	 * For all settings fields, the field slug, option sub key, and label
+	 * suffix have to match for the rendering in the callback to be
+	 * semantically correct.
+	 */
+	add_settings_field(
+		'benchmarking_actions',
+		__( 'Actions', 'performance-lab' ),
+		static function() {
+			perflab_render_server_timing_page_field( 'benchmarking_actions' );
+		},
+		PERFLAB_SERVER_TIMING_SCREEN,
+		'benchmarking',
+		array( 'label_for' => 'server_timing_benchmarking_actions' )
+	);
+	add_settings_field(
+		'benchmarking_filters',
+		__( 'Filters', 'performance-lab' ),
+		static function() {
+			perflab_render_server_timing_page_field( 'benchmarking_filters' );
+		},
+		PERFLAB_SERVER_TIMING_SCREEN,
+		'benchmarking',
+		array( 'label_for' => 'server_timing_benchmarking_filters' )
+	);
+}
+
+/**
+ * Renders the Server-Timing page.
+ *
+ * @since n.e.x.t
+ */
+function perflab_render_server_timing_page() {
+	?>
+	<div class="wrap">
+		<?php settings_errors(); ?>
+		<h1>
+			<?php esc_html_e( 'Server-Timing', 'performance-lab' ); ?>
+		</h1>
+
+		<form action="options.php" method="post" novalidate="novalidate">
```

Comment:
No need to use XHTML syntax as the admin is HTML5.

```suggestion
		<form action="options.php" method="post" novalidate>
```

But why `novalidate` in the first place? The UI could listen for `change` events on the `textarea` and then check for valid input to then call `setCustomValidity()` with `null` if it is valid or else an error message string which would then show up in the UI when the user tries to submit.

Reply:
I don't think the validation here is critical since it's sanitized later anyway, so implementing some validation would open a new topic of discussion on what it should be like. I don't think core's settings typically provide client-side validation, so I don't think we need to here either.

Regarding the XHTML syntax, I'll remove that.

Reply:
Done in https://github.com/WordPress/performance/pull/784/commits/a8d141f594672d3d1aa03e9b2e79b41d9cabdd0b

Reply:
Since there is no HTML validation constraints, then there's no need for the `novalidate` attribute at all. It's only applicable if you want to bypass the browser from doing client-side validation. But since none is present, it doesn't do anything.

Reply:
You're probably right, but aren't there some browser defaults that could still trigger client-side validation? I wonder why for example WP core then uses `novalidate` in `options-general.php`.

Reply:
It's most likely stems from the time (over a decade ago) that browsers implemented poor logic for validating email addresses: https://core.trac.wordpress.org/ticket/15080#comment:2

Since there is an `input[type=email]` on the general settings screen, it's probably why `novalidate` is added there. Nevertheless, the `novalidate` attribute is _absent_ from other settings screens, including: Writing, Reading, Discussion, Media, Permalinks, and Privacy. 

I also proposed removing `novalidate` from the comments form: https://core.trac.wordpress.org/ticket/47595

Reply:
Just checked and I see `novalidate` was only added to the forms that involve `input` fields of type `email` and `url`: https://github.com/WordPress/wordpress-develop/commit/0a4e8b2b7ee84efb28acc946a9a54cdd03414c28

Reply:
Updated in https://github.com/WordPress/performance/pull/784/commits/c48d206320a233ba5ef95a1c17780b1f8fbc0d9b